File: salauddin_sifat/219/main.go
package main

import "fmt"

func main() {
	result := containsNearbyDuplicate([]int{1, 2, 3, 1, 2, 3}, 2)
	fmt.Println(result)
}

func containsNearbyDuplicate(nums []int, k int) bool {
	memo := make(map[int]int)
	for i, v := range nums {
		if p, ok := memo[v]; ok && i-p <= k {
			return true
		}
		memo[v] = i
	}
	return false
}

File: salauddin_sifat/219/main.js
/**
 * 219. Contains Duplicate II
 */
let containsNearbyDuplicate = function (nums, k) {
  const memo = new Map();
  for (let i = 0; i < nums.length; i++) {
    const el = nums[i];
    if (Math.abs(memo.get(el) - i) <= k) return true;
    memo.set(el, i);
  }
  return false;
};

console.log(containsNearbyDuplicate([1, 2, 3, 1], 3));
